hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java

```diff
@@ -71,15 +72,43 @@ private boolean isSorted(List<Cell> list) {
         return true;
     }
 
+    private void handleSanityChecks(List<Cell> results,
+                                    List<Cell> rawList,
+                                    List<TxnView> txns) {
+        final boolean dataToReturnIsEmpty = dataToReturn.isEmpty();
+        final boolean resultsIsEmpty = results.isEmpty();
+        final boolean maxTombstoneIsNull = maxTombstone == null;
+        final boolean rawListAndTxnListSameSize = rawList.size() == txns.size();
+        final boolean debugSortCheck = !LOG.isDebugEnabled() || isSorted(rawList);
+
+        if (!debugSortCheck)
+            setBypassPurgeWithWarning("CompactionStateMutate: rawList is not sorted.");
+        if (!dataToReturnIsEmpty)
+            setBypassPurgeWithWarning("dataToReturn is not properly initialized.");
+        if (!resultsIsEmpty)
+            setBypassPurgeWithWarning("results list not properly initialized.");
+        if (!maxTombstoneIsNull)
+            setBypassPurgeWithWarning("maxTombstone not properly initialized to null.");
+        if (!rawListAndTxnListSameSize)
+            setBypassPurgeWithWarning("rawList and txn list not the same length.");
+
+        assert dataToReturnIsEmpty;
+        assert resultsIsEmpty;
+        assert maxTombstoneIsNull;
+        assert debugSortCheck : "CompactionStateMutate: rawList not sorted";
+        assert rawListAndTxnListSameSize;
+    }
+
+    private void setBypassPurgeWithWarning(String warningMessage) {
+        bypassPurge = true;
+        LOG.warn("Skipping tombstone purge.  " + warningMessage);
+    }
+
     /***
      * @return the size of all cells in the `rawList` parameter.
      */
     public long mutate(List<Cell> rawList, List<TxnView> txns, List<Cell> results) throws IOException {
-        assert dataToReturn.isEmpty();
-        assert results.isEmpty();
-        assert maxTombstone == null;
-        assert !LOG.isDebugEnabled() || isSorted(rawList) : "CompactionStateMutate: rawList not sorted";
-        assert rawList.size() == txns.size();
```

Comment:
We may want to keep assertions because they are fired during ITs and can catch bugs. In case they are disables on a cluster, handleSanity Checks can safely bypass purge.

Reply:
Looks like the same assertions were moved into the handleSanityChecks method, so they are being kept.

Reply:
They becomes warning messages and does not cause a failure for IT

Reply:
Feel free to write a Jira.  I'm merging now so it can get into this week's build.